README.md

```diff
@@ -1,48 +1,147 @@
 # terraform-registry-address
 
-This package helps with representation, comparison and parsing of
-Terraform Registry addresses, such as
-`registry.terraform.io/grafana/grafana` or `hashicorp/aws`.
+This module enables parsing, comparison and canonical representation of
+[Terraform Registry](https://registry.terraform.io/) **provider** addresses
+(such as `registry.terraform.io/grafana/grafana` or `hashicorp/aws`)
+and **module** addresses (such as `hashicorp/subnets/cidr`).
 
-The most common source of these addresses outside of Terraform Core
-is JSON representation of state, plan, or schemas as obtained
-via [`hashicorp/terraform-exec`](https://github.com/hashicorp/terraform-exec).
+**Provider** addresses can be found in
```

Comment:
Do we want to mention the [CLI configuration file](https://www.terraform.io/cli/config/config-file#provider-installation) and [reattach configurations](https://www.terraform.io/plugin/debugging#running-terraform-with-a-provider-in-debug-mode) as well?

Reply:
Added.